src/py/flwr/client/app.py

```diff
@@ -150,35 +152,41 @@ class `flwr.client.Client`.
     else:
         connection = grpc_connection
     while True:
-        sleep_duration: int = 0
-        with connection(
-            server_address,
-            max_message_length=grpc_max_message_length,
-            root_certificates=root_certificates,
-        ) as conn:
-            receive, send = conn
-
-            while True:
-                server_message = receive()
-                if server_message is None:
-                    time.sleep(3)  # Wait for 3s before asking again
-                    continue
-                client_message, sleep_duration, keep_going = handle(
-                    client, server_message
-                )
-                send(client_message)
-                if not keep_going:
-                    break
-        if sleep_duration == 0:
-            log(INFO, "Disconnect and shut down")
-            break
-        # Sleep and reconnect afterwards
-        log(
-            INFO,
-            "Disconnect, then re-establish connection after %s second(s)",
-            sleep_duration,
-        )
-        time.sleep(sleep_duration)
+        parsed_address = parse_address(server_address)
```

Comment:
Is there a reason for the address validation to happen inside the loop? If not, let's do it right after `event(...)`.